### PR TITLE
Fix bone/ghost dragon handling in skeleton transformer

### PR DESCRIPTION
**Pull request by dydzio0614**
Fixes mantis bug 2932

### CODE REVIEW

File: server/CGameHandler.cpp [hunk 1]
```diff
@@ -3668,7 +3668,9 @@ bool CGameHandler::transformInUndead(const IMarket *market, const CGHeroInstance
 	//resulting creature - bone dragons or skeletons
 	CreatureID resCreature = CreatureID::SKELETON;
 
-	if (s.hasBonusOfType(Bonus::DRAGON_NATURE)
+	if ((s.hasBonusOfType(Bonus::DRAGON_NATURE)
+			&& (s.getCreatureID() != CreatureID::BONE_DRAGON)
+			&& (s.getCreatureID() != CreatureID::GHOST_DRAGON))
```
Comment (alexvins) [inline server/CGameHandler.cpp]: This should be more generic solution f.e. `!hasBonusOfType(Bonus::UNDEAD)`

Comment (DjWarmonger) [reply server/CGameHandler.cpp]: We already have Dracolich for that matter

Comment (dydzio0614) [reply server/CGameHandler.cpp]: I do not see how is it much better - 1 less line of code? Current conditions make it fairly clear why they are here in the first place.

Comment (alexvins) [reply server/CGameHandler.cpp]: Condition should be: any dragon|hydra to bone dragon but undead dragon to skeleton.

**Pull request merged**